packages/sampler/stories/moonstone-stories/Button.js

```diff
@@ -9,11 +10,11 @@ import {withInfo} from '@storybook/addon-info';
 import {mergeComponentMetadata} from '../../src/utils/propTables';
 import nullify from '../../src/utils/nullify.js';
 
-const Config = mergeComponentMetadata('Button', ButtonBase, Button);
+const Config = mergeComponentMetadata('Button', ButtonBase, Button, UIButton, UIButtonBase);
 
 // Set up some defaults for info and knobs
 const prop = {
-	backgroundOpacity: ['', 'translucent', 'lightTranslucent', 'transparent'],
+	backgroundOpacity: ['opaque', 'translucent', 'lightTranslucent', 'transparent'],
```

Comment:
Having no blank (`''`) option implies that the component must use this prop. `backgroundOpacity` is optional and the sample should represent that. I'm fine with _also_ having a blank/opaque option, but not ok with replacing the blank with 'opaque'.

Reply:
Similar to the comment below, it seemed awkward to me when Button's `backgroundOpacity` has `@default 'opaque` in the documentation, but not set anything in the sampler. Plus, it is documented as one of the options but the user cannot select it in the sampler. I thought that was inconsistent. Thoughts?

Reply:
I agree. Let's remove the `@default 'opaque'` from the docs. It was put there as a way to explain that the "normal" state of a Button's appearance is opaque, and not translucent, so PropTypes wouldn't error if someone attempted to use a logical word for an opacity property, even though it's just, and already, _the default_.

`''` in this case, due to `nullify`, below removes `backgroundOpacity` from the JSX entirely, so it appears as `<Button>` rather than implying there's a required property: `<Button backgroundOpacity="opaque">`.

We're getting into "developer experience" (DX) territory here, rather than discussing `minWidth`. A fundamental question like "should an empty prop value be enumerated in an array of valid options so it's clear to a developer that they can specify nothing and get the default on a non-required property". My preference is "no", because as a developer I want to see what I'm _practically_ able to set a prop to, not an exhaustive set of values. (We don't list `null` or `undefined` as options, even though those technically also work.) If I choose to not set it in my app, I'll just exclude that prop from my code and expect it to work. I definitely don't want developers to think they must assign an empty value: `<Button backgroundOpacity="">` to make a simple button work. That being said, we do have to sort of play nicely with Storybook stories, and massage them into accurately representing our options. There is no way to temporarily _remove_ a knob from the Knobs list, so we need a way to indicate, in our existing `select` knob, that not providing a prop is a valid option. This is where the `''` comes in (as the best, most representative option we've found so far). It's necessary to show that not choosing a value is an acceptable option.

"opaque" was not listed as a knob value because I didn't want to cause confusion about what the values mean. If `''` and "opaque" are both present, it could imply that they should visually represent two different states, i.e. whatever the default/blank is, and opaque, which must be different, because it's listed as a separate option.

If we **must** change the documentation, PropTypes, and Story, I'd personally prefer to remove the `@default 'opaque'` and leave the `''` in the knob values. The documentation will also need to be updated to describe that Button, by default, is visually opaque but that an "opaque" string value is not a valid option for the `backgroundOpacity` prop, so it's clear what is going on.

Reply:
I agree. Let's remove the `@default 'opaque'` from the docs. It was put there as a way to explain that the "normal" state of a Button's appearance is opaque, and not translucent, so PropTypes wouldn't error if someone attempted to use a logical word for an opacity property, even though it's just, and already, _the default_.

`''` in this case, due to `nullify`, below removes `backgroundOpacity` from the JSX entirely, so it appears as `<Button>` rather than implying there's a required property: `<Button backgroundOpacity="opaque">`.

We're getting into "developer experience" (DX) territory here, rather than discussing `minWidth`. A fundamental question like "should an empty prop value be enumerated in an array of valid options so it's clear to a developer that they can specify nothing and get the default on a non-required property". My preference is "no", because as a developer I want to see what I'm _practically_ able to set a prop to, not an exhaustive set of values. (We don't list `null` or `undefined` as options, even though those technically also work.) If I choose to not set it in my app, I'll just exclude that prop from my code and expect it to work. I definitely don't want developers to think they must assign an empty value: `<Button backgroundOpacity="">` to make a simple button work. That being said, we do have to sort of play nicely with Storybook stories, and massage them into accurately representing our options. There is no way to temporarily _remove_ a knob from the Knobs list, so we need a way to indicate, in our existing `select` knob, that not providing a prop is a valid option. This is where the `''` comes in (as the best, most representative option we've found so far). It's necessary to show that not choosing a value is an acceptable option.

"opaque" was not listed as a knob value because I didn't want to cause confusion about what the values mean. If `''` and "opaque" are both present, it could imply that they should visually represent two different states, i.e. whatever the default/blank is, and opaque, which must be different, because it's listed as a separate option.

If we **must** change the documentation, PropTypes, and Story, I'd personally prefer to remove the `@default 'opaque'` and leave the `''` in the knob values. The documentation will also need to be updated to describe that Button, by default, is visually opaque but that an "opaque" string value is not a valid option for the `backgroundOpacity` prop, so it's clear what is going on.